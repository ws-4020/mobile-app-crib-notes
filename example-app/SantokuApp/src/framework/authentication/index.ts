export {useAuthenticationService, ActiveAccountIdNotFoundError, PasswordNotFoundError} from './AuthenticationService';
