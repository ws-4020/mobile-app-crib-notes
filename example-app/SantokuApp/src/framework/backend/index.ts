export * from './WithReactQuery';
export * from './useDefaultOptions';
export {refreshCsrfToken} from './useCustomInstance';
