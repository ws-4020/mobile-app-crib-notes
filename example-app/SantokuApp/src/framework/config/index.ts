import {config} from './Config';

export {config};
