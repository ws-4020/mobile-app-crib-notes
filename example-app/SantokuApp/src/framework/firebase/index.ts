import {firebaseConfig} from './FirebaseConfig';

export {firebaseConfig};
