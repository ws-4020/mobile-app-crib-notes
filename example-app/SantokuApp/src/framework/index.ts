export {AuthnService, ActiveAccountIdNotFoundError, PasswordNotFoundError} from './authentication';
export {api, csrfToken} from './backend';
export {AppConfig} from './config';
export {firebaseConfig, FirebaseConfig} from './firebase';
export type {Logger, LoggerOptions} from './logging';
export {log, createLogger} from './logging';
export {m, loadMessages, BundledMessagesLoader} from './message';
export {launchedId, generatePassword, useIsMounted} from './utilities';
