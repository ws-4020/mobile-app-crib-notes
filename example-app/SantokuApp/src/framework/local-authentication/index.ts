import {ExpoLocalAuthentication} from './ExpoLocalAuthentication';
import {LocalAuthentication} from './LocalAuthentication';

export * from './LocalAuthentication';

const localAuthentication: LocalAuthentication = new ExpoLocalAuthentication();

export default localAuthentication;
