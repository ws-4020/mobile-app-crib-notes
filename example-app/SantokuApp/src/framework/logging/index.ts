import {FirebaseCrashlyticsTransport} from './FirebaseCrashlyticsTransport';
import {Logger, createLogger, DEFAULT_LOGGER_OPTIONS, LoggerOptions} from './Logger';

const log = __DEV__
  ? createLogger(DEFAULT_LOGGER_OPTIONS)
  : createLogger({...DEFAULT_LOGGER_OPTIONS, transport: new FirebaseCrashlyticsTransport()});

export type {Logger, LoggerOptions};
export {log, createLogger};
