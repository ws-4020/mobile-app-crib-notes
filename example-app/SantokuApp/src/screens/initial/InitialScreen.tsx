import {StackActions, useNavigation} from '@react-navigation/native';
import {useAppTheme} from 'components/theme';
import * as SplashScreen from 'expo-splash-screen';
import {m} from 'framework';
import {log} from 'framework/logging';
import {useIsMounted} from 'framework/utilities/useIsMounted';
import React, {useCallback, useEffect} from 'react';
import {ActivityIndicator, Alert, StyleSheet, View} from 'react-native';
import RNExitApp from 'react-native-exit-app';

const ScreenName = 'InitialScreen';
const Screen: React.FC = () => {
  const navigation = useNavigation();
  const isMounted = useIsMounted();
  const appTheme = useAppTheme();

  const prepare = useCallback(async () => {
    // TODO: 利用規約同意状態取得APIを呼び出す処理に置き換え
    const hasAgreedValidTermsOfService = await new Promise((resolve) => resolve(false));
    if (!hasAgreedValidTermsOfService) {
      navigation.dispatch(StackActions.replace('TermsOfServiceAgreement'));
    } else {
      navigation.dispatch(StackActions.replace('AuthenticatedStackNav'));
    }
    // スプラッシュスクリーンを閉じる前に100ms程度は待たないと、コンポーネントのレンダリングが完了せず画面が一瞬白くなってしまいます。
    // expo-app-loadingでも200ms待っているので、ここでも200ms待つようにしています。
    // cf: https://github.com/expo/expo/blob/sdk-41/packages/expo-app-loading/src/AppLoadingNativeWrapper.tsx#L27
    await new Promise((resolve) => setTimeout(resolve, 200));
    try {
      await SplashScreen.hideAsync();
    } catch (e) {
      // 既にSplashScreenが消えていた場合はそのまま継続
    }
  }, [navigation]);

  const onError = useCallback(() => {
    Alert.alert(m('初期化エラー'), m('app.初期化エラー'), [
      {
        text: 'OK',
        style: 'default',
        onPress: () => {
          RNExitApp.exitApp();
        },
      },
    ]);
  }, []);

  useEffect(() => {
    prepare().catch(() => {
      log.error('Failed to load initial data.', 'InitialDataLoadingFailure');
      if (isMounted()) {
        onError();
      }
    });
  }, [prepare, isMounted, onError]);

  return (
    <View style={styles.container} testID="InitialScreen">
      <ActivityIndicator size="large" color={appTheme.colors.primary} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export const InitialScreen = {
  component: Screen,
  name: ScreenName,
};
