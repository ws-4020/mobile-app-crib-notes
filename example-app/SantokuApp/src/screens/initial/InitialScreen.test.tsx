import {StackActions} from '@react-navigation/routers';
import '@testing-library/jest-native/extend-expect';
import {render, waitFor} from '@testing-library/react-native';
import {BundledMessagesLoader, loadMessages} from 'framework';
import React from 'react';

import {InitialScreen} from './InitialScreen';

beforeAll(async () => {
  await loadMessages(new BundledMessagesLoader());
});

beforeEach(() => {
  jest.useFakeTimers();
});

describe('InitialScreen', () => {
  it('マウントされたときに正常にレンダリングされること', () => {
    const app = render(<InitialScreen.component />);
    expect(app.queryByTestId('InitialScreen')).not.toBeNull();
    expect(app).toMatchSnapshot();
  });

  it('利用規約未同意の場合に利用規約画面へ遷移すること', async () => {
    render(<InitialScreen.component />);
    await waitFor(() => {
      expect(__mocks.navigation.dispatch).toHaveBeenCalledWith(StackActions.replace('TermsOfServiceAgreement'));
    });
  });

  // TODO: 利用規約同意済みの場合にホーム画面へ遷移することのテスト(API呼び出し対応後)
  // TODO: 利用規約同意状態の取得に失敗した場合にエラーダイアログが表示され、OKを押すとアプリが終了すること(API呼び出し対応後)

  it('処理完了後200ms経ってからスプラッシュスクリーンが閉じられること', async () => {
    render(<InitialScreen.component />);
    expect(__mocks.expoSplashScreen.hideAsync).not.toHaveBeenCalled();
    jest.advanceTimersByTime(200);
    await waitFor(() => {
      expect(__mocks.expoSplashScreen.hideAsync).toHaveBeenCalled();
    });
  });
});
