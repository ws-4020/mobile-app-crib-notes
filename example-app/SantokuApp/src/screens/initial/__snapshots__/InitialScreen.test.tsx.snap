// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialScreen マウントされたときに正常にレンダリングされること 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="InitialScreen"
>
  <ActivityIndicator
    animating={true}
    color="#3a6be8"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
